### ISSUE TITLE
Use Dependent Resources from the Operator SDK

### ISSUE THREAD

**Issue by vmuzikar** [keycloak/keycloak#13453]
We're currently managing Operator-owned resources with a [custom logic](https://github.com/keycloak/keycloak/blob/7e925bfbff5410d27675f10d596b752fa794a4ab/operator/src/main/java/org/keycloak/operator/controllers/OperatorManagedResource.java). We should migrate to [Dependent Resources](https://javaoperatorsdk.io/docs/dependent-resources) coming from the SDK.

Should be done after #13451.

Tasks in this epic:
- [x] #22182
- [x] #22183
- [x] #22206
- [x] #22207
- [x] #22223
- [x] #22225

**Comment by shawkins** [comment]
@vmuzikar I don't seem to have rights to add tasks.  I'll start linking them in here: #22182

**Comment by shawkins** [comment]
@vmuzikar do you want the RealmImport logic as a part of this?  Or did I understand correctly that the mechanism may change to just a Realm CR?

**Comment by vmuzikar** [comment]
The Realm Import CR will _probably_ get deprecated in some time in favor of a proper Realm CR. But it's a possibility it'll remain as Realm CR definitely won't cover full Realm representation as the Import CR does. Hence, IMHO Realm Import should be converted to dependent resource as well if it's not too much effort.

**Comment by shawkins** [comment]
Having nearly completed the conversion there are a coupld of thoughts / additional tasks:
- It will be good to pickup the fix for https://github.com/operator-framework/java-operator-sdk/pull/1994 - that should happen along with the logging changes and the fix to make the bundle logic build time only
- There are several places not using a cache where one may be applicable (secrets, statefulsets).  This can be a lower priority task as we should already be making signifcantly fewer calls to the api server.
- We can consider changing the realmimport logic to be based upon keycloaks, rather than statefulsets.
- We may still want something like the MetadataAwareOnUpdateFilter in place to prevent vacuous reconciliation loops (there shouldn't be any that we're currently hitting, but they could creep back in)

**Comment by shawkins** [comment]
Marking as resolved.

**Comment by marchermans** [comment]
I would like to re-open this ticket. As resources like the Statefulset for KeyCloak do not have this handling properly implemented.

Should I open a new ticket, or do you want to reopen this ticket?

**Comment by vmuzikar** [comment]
> As resources like the Statefulset for KeyCloak do not have this handling properly implemented.

Feel free to open a new ticket if there's a bug. Thanks.
